application/classes/Model/User.php

```diff
@@ -192,14 +192,18 @@ public static function all() {
 		return (new Model_User())->find_all();
 	}
 	
+	public static function valid() {
+		return (new Model_user())->where('email', 'NOT LIKE', 'customer%')->find_all();
+	}
+	
 	/**
 	 * Retrieve all convention registrants (i.e. users who registered for a convention)
 	 * @param Model_Convention $con Convention to retrieve users for
 	 * @return Generator list of users
 	 */
 	public static function byConvention(Model_Convention $con) {
 		$seen = [];
-		foreach ((new Model_Ticket())->with('sale')->where('convention_id', '=', $con->pk())->find_all() as $ticket) {
+		foreach ((new Model_Sale())->where('convention_id', '=', $con->pk())->find_all() as $ticket) {
 			if (@$seen[$ticket->user->pk()]) continue;
 			$seen[$ticket->user->pk()] = 1;
 			yield $ticket->user;
```

Comment:
was playing with new syntax in 7 :-) maybe not appropriate because now there are different ways to handle similar things, but I couldn't resist. I'd probably kill it at a later date.